src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -172,7 +172,22 @@ public void deleteApplication(String project, String application) throws IOExcep
 
     @Override
     public void push(String project, String application, String context, String component) throws IOException {
-        ExecHelper.executeWithTerminal(new File(context), envVars, command, "push");
+        push(context, false);
+    }
+
+    @Override
+    public void pushWithDebug(String project, String application, String context, String component) throws IOException {
+        push(context, true);
+    }
+
+    private void push(String context, boolean debug) throws IOException {
+        List<String> args = new ArrayList<>();
+        args.add(command);
+        args.add("push");
+        if (debug) {
+            args.add("--debug");
+        }
+        ExecHelper.executeWithTerminal(new File(context), envVars, args.toArray(new String[0]));
```

Comment:
Seems to me that there is a deadlock: because this is called from a modal dialog, the terminal window is never created thus the process is blocked because nobody is reading its output.
So my while IJ is stuck, clicking cancel does nothing and the only way to get out is to kill the process using the Windows Process Manager (where I can see odo push --debug being launched but never returns)